scripts/generate_example_pngs.R

```diff
@@ -0,0 +1,191 @@
+
+    if (is.null(resize)) {
+      webshot::webshot(
+        url = paste0("file:///", tempfile_),
+        file = filename,
+        zoom = zoom,
+        expand = expand,
+        ...
+      )
+    } else {
+      webshot::webshot(
+        url = paste0("file:///", tempfile_),
+        file = filename,
+        zoom = zoom,
+        expand = expand,
+        ...
+      ) %>%
+        webshot::resize(resize)
+    }
+  }
+}
+
+# Declare a fixed image width
+image_width <- 1100
+
+# Obtain a temporary directory name
+temp_dir <- tempdir()
+
+# Generate images in `temp_dir`
+for (fn_name in fn_names) {
+
+  # Get the initial list of objects in the environment
+  obj_list_initial <- ls()
+
+  # Evaluate the examples in the examples for each topic/function; the
+  # objects we'd want to create images from all begin with `tab_`
+  utils::example(topic = fn_name, package = "gt", character.only = TRUE, give.lines = FALSE)
+
+  # Get a vector of new objects created
+  new_obj <- base::setdiff(ls(), obj_list_initial)
+
+  # Get a separate vector of objects that begin with `tab_`
+  tab_obj <- new_obj %>% grep("^tab_.\\d*?", ., value = TRUE)
+
+  # For each of the `tab_` objects, create an image using webshot
+  for (i in seq_along(tab_obj)) {
+
+    # Create a systematic file name so that the help file
+    # can access multiple images in the correct order
+    filename <- file.path(temp_dir, paste0("man_", fn_name, "_", i, ".png"))
+
+    # Save the image with a high `zoom` value for clarity,
+    # but resized down to 50% of the original
+    save_with_webshot(
+      data = get(tab_obj[i]),
+      filename = filename,
+      zoom = 3,
+      selector = "table",
+      resize = "50%"
+    )
+
+    # Get the width and height of the image produced by webshot
+    image_x <- readPNG(filename) %>% dim() %>% .[2]
+    image_y <- readPNG(filename) %>% dim() %>% .[1]
+
+    if (image_x > image_width) {
+      eff_image_x <- image_x
+    } else {
+      eff_image_x <- image_width
+    }
+
+    if (image_y >= 1500) {
+      eff_image_y <- 1500
+    } else {
+      eff_image_y <- image_y
+    }
+
+    # Use ImageMagick (installed on system) to widen all images to 1100px
+    system(glue::glue("convert {filename} -gravity center -extent {eff_image_x}x{eff_image_y} {filename}"))
+  }
+
+  # Clean up some of the created objects
+  rm(list = new_obj)
+  rm(new_obj)
+  rm(tab_obj)
+}
+
+# Copy all PNG files to a clean temporary directory
+temp_dir_final <- paste0(tempdir(), "final")
+dir.create(temp_dir_final)
+png_list <- list.files(temp_dir, ".*\\.png$")
+file.copy(file.path(temp_dir, png_list), file.path(temp_dir_final, png_list))
+
+# Shrink PNG files with `webshot::shrink()`
+webshot::shrink(filename = file.path(temp_dir_final, png_list))
+
+# Use `pngquant` to losslessly compress PNG files
+# The project is at https://github.com/kornelski/pngquant
+system(glue::glue("cd {temp_dir_final}; pngquant --force --ext=.png --skip-if-larger --nofs --speed 1 *.png"))
```

Comment:
This might not work if `temp_dir_final` has spaces in it; maybe do

```r
old_dir <- setwd(temp_dir_final)
on.exit(setwd(old_dir), add = TRUE)
system("pngquant ...")
```

Reply:
Used `shQuote()` for paths in all `system()` calls.